src/api/store/community.py

```diff
@@ -1,16 +1,206 @@
+          # zip code already used
+          message = 'Zipcode %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'CITY' and location.city == loc:
+          message = 'City %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'COUNTY' and location.county == loc:
+          message = 'County %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'STATE' and location.state == loc:
+          message = 'State (%s) is already part of another geographic community (%s).' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'COUNTRY' and location.country == loc:
+          message = 'Country (%s) is already part of another geographic community (%s).' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+
+  def _update_locations(self, geography_type, location_set, community):
+        
+    # clean up in case there is garbage in there
+    if community.location_set:
+      community.location_set.clear()
+
+    # this is a list of zipcodes, towns, cities, counties, states        
+    location_list = location_set.replace(", ",",").split(",")  # passed as comma separated list
+    print("Community includes the following locations :"+str(location_list))
+    for location in location_list:
+      if geography_type == 'ZIPCODE':
+        if location[0].isdigit():
+          location = location.replace(" ","")
+
+          # looks like a zipcode.  Check which towns it corresponds to
+          zipcode = zipcodes.matching(location)
+          if len(zipcode)>0:
+            city = zipcode[0].get('city', None)  
+          else:
+            raise Exception('No zip code entry found for zip='+location)
+
+          loc, created = Location.objects.get_or_create(location_type='ZIP_CODE_ONLY', zipcode=location, city=city)
+          if created:
+            print("Zipcode "+location+" created for town "+city)
+          else:
+            print("Zipcode "+location+" found for town "+city)
+                        
+          self._check_geography_unique(community, geography_type, location)
+
+        else:
+          # assume this is a town, see we can find the zip codes associated with it
+          ss = location.split('-')
+          town = ss[0]
+          if len(ss)==2:
+            state = ss[1]
+          else:
+            state = 'MA'
+            
+          zips = zipcodes.filter_by(city=town, state=state, zip_code_type='STANDARD' )
+          print("Number of zipcodes = "+str(len(zips)))
+          if len(zips)>0:
+            for zip in zips:
```

Comment:
very expensive move

Reply:
You've got to be kidding me.  This isn't doing database access, it is accessing an array.  Tell me why this is expensive.  Again, this is run by a Super admin once in a blue moon.